nav2_map_server/src/map_io.cpp

```diff
@@ -90,7 +90,7 @@ char * dirname(char * path)
     /* This assignment is ill-designed but the XPG specs require to
        return a string containing "." in any case no directory part is
        found and so a static and constant string is required.  */
-    path = reinterpret_cast<char *>(dot);
+    path = (char *)dot;
```

Comment:
Revert or find another way to do it in a modern C++ way

Reply:
This one definitely threw a compiler error saying it couldn't do the conversion. My only guess was dot being const? 

Otherwise char[2] and char* seem pretty similar.

Reply:
I guess then it should be a const_cast, not a reinterpeter_cast